webapp/src/ts/components/search-bar/search-bar.component.html

```diff
@@ -1,5 +1,8 @@
 <section class="mm-search-bar-container" [class.open-search]="openSearch" [class.disabled]="disabled">
   <div class="search-bar-left-icon">
+    <label for="barcode-scanner-input" class="fa fa-qrcode" *ngIf="isBarcodeScannerAvailable && showSearchIcon()" (click)="onBarcodeOpen()">
+      <input type="file" id="barcode-scanner-input" data-type-xml="binary" accept="image/*" (change)="processBarcodeFile($event)"/>
+    </label>
```

Comment:
```suggestion
    <label class="fa fa-qrcode" *ngIf="isBarcodeScannerAvailable && showSearchIcon()" (click)="onBarcodeOpen()">
      <input type="file" id="barcode-scanner-input" data-type-xml="binary" accept="image/*" (change)="processBarcodeFile($event)"/>
    </label>
```

you don't need to pass the `<input />`'s id to the `<label />`'s `for` property if the input is a child element of the label

Reply:
Nice!